### PR TITLE
fix 195

### LINKED ISSUE #195: [bug] ListTaskInstanceParameter does not move when not specific `requires` method
```python
import gokart

class Piyo(gokart.TaskOnKart):
    def run(self):
        self.dump('piyo')

class Hoge(gokart.TaskOnKart):
    hoge = gokart.ListTaskInstanceParameter()

    def run(self):
        hoge = self.load_data_frame('hoge')


gokart.build(Hoge(hoge=[Piyo(), Piyo()]))
```

```
Traceback (most recent call last):
  File "python3.7/site-packages/luigi/worker.py", line 191, in run
    new_deps = self._run_get_new_deps()
  File "python3.7/site-packages/luigi/worker.py", line 133, in _run_get_new_deps
    task_gen = self.task.run()
  File "hoge.py", line 17, in run
    df = self.load_data_frame('hoge')
  File "python3.7/site-packages/gokart/task.py", line 236, in load_data_frame
    dfs = self.load(target=target)
  File "python3.7/site-packages/gokart/task.py", line 211, in load
    return _load(self._get_input_targets(target))
  File "python3.7/site-packages/gokart/task.py", line 276, in _get_input_targets
    return self.input()[target]
TypeError: list indices must be integers or slices, not str
```

### PR DESCRIPTION
**Pull request by vaaaaanquish**
fix #195

### CODE REVIEW

**Comment by vaaaaanquish** [comment]
@Hi-king @hirosassa @mski-iksm plz review

File: gokart/task.py [hunk 1]
```diff
@@ -77,7 +77,10 @@ def requires(self):
         return tasks or []  # when tasks is empty dict, then this returns empty list.
 
     def make_task_instance_dictionary(self) -> Dict[str, 'TaskOnKart']:
```
Comment (hirosassa) [inline gokart/task.py:79]: I would like you to write test(s) on `mae_task_instance_dictionary`

**Review by hirosassa — CHANGES_REQUESTED** [review]
Thanks for your bug report and good PR! I added a comment.

File: gokart/task.py [hunk 2]
```diff
@@ -77,7 +77,10 @@ def requires(self):
...
-        return {key: var for key, var in vars(self).items() if isinstance(var, TaskOnKart)}
+        return {key: var for key, var in vars(self).items() if isinstance(var, TaskOnKart) or self._is_list_of_task_on_kart(var)}
+
+    def _is_list_of_task_on_kart(self, x):
```
Comment (Hi-king) [inline gokart/task.py:82]: [IMO] 
I think it's better to include ` isinstance(var, TaskOnKart)` into `_is_list_of_task_on_kart ` .
(And of course rename the method)

File: gokart/task.py [hunk 3]
```diff
@@ -77,7 +77,10 @@ def requires(self):
...
+        return isinstance(x, list) and x and isinstance(x[0], TaskOnKart)
```
Comment (Hi-king) [inline gokart/task.py:83]:
```suggestion
        return isinstance(x, list) and x and all([isinstance(item, int) for item in x])
```

Though it's not a big problem, it's better to check all items :)

File: gokart/task.py (hunk 2 above)
Comment (hirosassa) [reply gokart/task.py:82]: ++

Comment (vaaaaanquish) [reply gokart/task.py:82]: Alright:)

**Comment by vaaaaanquish** [comment]
@Hi-king @hirosassa @mski-iksm plz review

File: test/test_task_on_kart.py [hunk 4]
```diff
@@ -471,6 +471,12 @@ def _wrap(func):
             task.run()
             mock_obj.assert_not_called()
 
+    def test_is_task_on_kart(self):
+        self.assertEqual(True, gokart.TaskOnKart.is_task_on_kart(gokart.TaskOnKart()))
+        self.assertEqual(False, gokart.TaskOnKart.is_task_on_kart(1))
+        self.assertEqual(False, gokart.TaskOnKart.is_task_on_kart(list()))
+        self.assertEqual(True, gokart.TaskOnKart.is_task_on_kart([gokart.TaskOnKart(), gokart.TaskOnKart()]))
+
```
Comment (hirosassa) [inline test/test_task_on_kart.py:479]: 👍

**Review by hirosassa — APPROVED** [review]
LGTM!

**Comment by Hi-king** [comment]
LGTM!

**Review by Hi-king — APPROVED** [review]

**Pull request merged**